rfcs/toast-component.md

```diff
@@ -0,0 +1,116 @@
+# Toast Notification Component
+
+Toast notifications are used to display a brief alert, warning, or confirmation message or a banner.
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Toast} from 'baseui/toast';
+
+export default () => <Toast>This is a notification message.</Toast>;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {Toast} from 'baseui/toast';
+
+export default () => {
+  return (
+    <React.Fragment>
+      <Toast
+        onClose={()=>{console.log('Toast notification was dismissed.')}}
+        autoHideDuration={5000}
+        overrides={{
+          Bosy: props => <CustomContainer>{props.children}</CustomContainer>,
+        }}
+      >
+      This a dismissable info notification that is hidden after 5 seconds automatically.
+      </Toast>
+      <Toast
+        closable={false}
+        kind={KIND.error}
+      >
+      {({dismiss}) => {
+        return (
+          <div>
+            This is an error notification with an action button
+            <button onClick={() => {
+              dismiss();
+              console.log('Action button clicked');
+            }}>
+              Take action
+            </button>
+          </div>
+        );
+      }}
+      </Toast>
+    </React.Fragment>
+  );
+}
+```
+
+## Exports
+
+* `Toast`
+* `StyledRoot`
+* `StyledBody`
+* `StyledCloseButton`
+* `KIND`
+* `PLACEMENT`
+
+## `Toast` API
+
+* `children: React.node | ({dismiss}) => React.node` - Required
+  Toast notification content. The children-as-function receives a `dissmiss` method that can be called to dismiss the notification and can be used as a handler for an action inside the toast content
+* `closable: boolean` - Optional. Defaults to true
+  When set to true a close button is displayed and the notification can be dismissed by a user
+* `kind: 'info' | 'success' | 'warning' | 'error'` - Optional. Defaults to 'info'
+  Defines the type of notification
+* `onClose: function` - Optional
+  A callback function called when a notification is dismissed
+* `overrides: {Root, Body, CloseButton}` - Optional
+  Overrides for presentational components. See "Presentational Components Props API" below.
+  * `[ComponentName]: ReactComponent | {props: {}, style: {}, component: ReactComponent}` - Optional
+* `placement: 'inline' | 'topLeft' | 'top' | 'topRight' | 'bottomRight' | 'bottom' | 'bottomLeft'` - Optional. Defaults to 'top'
+  Position of a toast notification relative to the browser window. If set to 'inline' a notitication is positioned according to a normal flow of the document
+* `autoHideDuration: number` - Optional. Defaults to 0
```

Comment:
we can call just `duration`.

Reply:
I'd say `autoHideDuration` gives more context on its usage. But happy to consider other options or accept this one if it gets more support.